src/common/interpolation.c

```diff
@@ -702,7 +702,7 @@ void dt_interpolation_compute_pixel4c(const struct dt_interpolation *itor,
     }
 
     for_each_channel(c,aligned(out))
-      out[c] = oonorm * pixel[c];
+      out[c] = fmaxf(0.0f, oonorm * pixel[c]);
```

Comment:
fmaxf() often does not vectorize, while the MAX() macro does.  That makes the latter preferable unless NaNs or infinities need to be handled.

Reply:
This is in fact about avoiding NaNs.

I will investigate this further and report back